### ISSUE TITLE
NetCoreApp2.1 using SDK 2.1.1 package downgrade to 2.1.0

### ISSUE THREAD

**Issue by Petermarcu** [dotnet/cli#9624]
@grantwar commented on [Wed Jul 04 2018](https://github.com/dotnet/core/issues/1754)

# Service Fabric Service using SDK 2.1.1 has an associated test core 2.1 project which is showing up as using SDK 2.1.0, won't allow me to upgrade via nuget (Blocked by project). 
# Build Error

> Detected package downgrade: Microsoft.NETCore.App from 2.1.1 to 2.1.0. Reference the package directly from the project to select a different version.

# Issue
I'm unable to get my solution to build due to an issue with a unit test project which references my service project forcing a package downgrade of my service project from 2.1.1 down to 2.1.0
No where have I explicitly set the dotnet core revision, and my netcoreapp projects are all 2.1
Prior to me installing dotnetcore 2.1.301, I had the 2.1.3 (RC1) and everything worked just fine.
I'm unable to manually change the version of the unit test project from 2.1.0 to 2.1.1, and doing doing a  install package microsoft.netcore.app -version 2.1.1 does not fix the issue either.

I can set `<TargetLatestRuntimePatch>false</TargetLatestRuntimePatch>` in the service project, and this does resolve the problem, however I'm not happy having to do this in a production service which is being worked on by a team of developers.
# Setup
dotnet --version 
> 2.1.301

Service fabric service project
![image](https://user-images.githubusercontent.com/8554565/42284016-1e4c1910-7fa3-11e8-8dda-216f6d6206c2.png)

Service unit test project
![image](https://user-images.githubusercontent.com/8554565/42284129-788ac034-7fa3-11e8-9597-a67ab7b16712.png)

Service unit test project nuget package manager
![image](https://user-images.githubusercontent.com/8554565/42284200-b0b93ad0-7fa3-11e8-9c08-51a8caee2dbd.png)

---

@agr commented on [Wed Jul 04 2018](https://github.com/dotnet/core/issues/1754#issuecomment-402601980)

Seeing the same in my project with xUnit when specifying runtime when building. Repro:

```
> dotnet --version
2.1.301
```

```
mkdir 211_210
cd 211_210
mkdir mvc
cd mvc
dotnet new mvc
cd ..
mkdir test
cd test
dotnet new xunit
dotnet add reference ../mvc/mvc.csproj
dotnet build -r any
```

Produces the following output:

```
Microsoft (R) Build Engine version 15.7.179.6572 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restoring packages for C:\Users\agr\Source\Repos\211_210\test\test.csproj...
  Restoring packages for C:\Users\agr\Source\Repos\211_210\mvc\mvc.csproj...
  Restore completed in 46.55 ms for C:\Users\agr\Source\Repos\211_210\test\test.csproj.
C:\Users\agr\Source\Repos\211_210\test\test.csproj : error NU1605: Detected package downgrade: Microsoft.NETCore.App from 2.1.1 to 2.1.0. Reference the package directly from the project to select a different version.
C:\Users\agr\Source\Repos\211_210\test\test.csproj : error NU1605:  test -> mvc -> Microsoft.NETCore.App (>= 2.1.1)
C:\Users\agr\Source\Repos\211_210\test\test.csproj : error NU1605:  test -> Microsoft.NETCore.App (>= 2.1.0)
  Restore failed in 294.47 ms for C:\Users\agr\Source\Repos\211_210\test\test.csproj.
  Restore completed in 629.54 ms for C:\Users\agr\Source\Repos\211_210\mvc\mvc.csproj.

Build FAILED.

C:\Users\agr\Source\Repos\211_210\test\test.csproj : error NU1605: Detected package downgrade: Microsoft.NETCore.App from 2.1.1 to 2.1.0. Reference the package directly from the project to select a different version.
C:\Users\agr\Source\Repos\211_210\test\test.csproj : error NU1605:  test -> mvc -> Microsoft.NETCore.App (>= 2.1.1)
C:\Users\agr\Source\Repos\211_210\test\test.csproj : error NU1605:  test -> Microsoft.NETCore.App (>= 2.1.0)
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:01.03
```

Just `dotnet build` seems to work, but if any runtime is specified it fails. I tried `any`, `win`, `win-x64`, `linux-x64`. Same happens both in Linux and Windows. It worked (and still works) fine in `2.1.300`.

---

@rrelyea commented on [Fri Jul 06 2018](https://github.com/dotnet/core/issues/1754#issuecomment-403092600)

This probably requires a fix in the templates or the SDK? (SDK team?)

MVC wants 2.1.1 of NetCoreApp
Unit test projects wants 2.1.0 of NetCoreApp.

Normally, this is a NuGet warning, but SDK projects have turned NU1605 into an error on purpose (Warning as Error).

---

@rrelyea commented on [Fri Jul 06 2018](https://github.com/dotnet/core/issues/1754#issuecomment-403096820)

My guess is that this should be closed, in favor of the following info I got from @dsplaisted in email:

This is part of a class of issues we have.  If a RID is specified, and the project is an Exe, then by default we’ll use the latest patch of Microsoft.NETCore.App.  We run into issues when you reference a project that uses the latest patch from one that doesn’t.

Other related issues:
https://github.com/dotnet/sdk/issues/2364
https://github.com/dotnet/sdk/issues/2253

**Comment by peterhuene** [comment]
Assigning this to 2.2.1 to see if we can do a better job with these exe project references regarding `TargetLatestRuntimePatch`.

**Comment by dsplaisted** [comment]
I believe this will be fixed in 2.1.100 by https://github.com/dotnet/sdk/pull/2395.

Until then, an alternative workaround which doesn't involve modifying the production project is to put `<TargetLatestRuntimePatch>true</TargetLatestRuntimePatch>` in your test project.

**Comment by replaysMike** [comment]
`<TargetLatestRuntimePatch>true</TargetLatestRuntimePatch>` does not work to solve this issue.

Only `<RuntimeFrameworkVersion>2.1.2</RuntimeFrameworkVersion>` works.

**Comment by fulee85** [comment]
I fixed this problem with Nuget Packages Manager. In the Browse tab search for Microsoft.NETCore.App. If you select the first result you can update it to the needed version.

**Comment by jdstuart** [comment]
I wish there was a way to upvote certain issues. Just ran into the same issue.

**Comment by Scandal-UK** [comment]
This is awful, none of the workarounds fix the issue for me and neither my WebAPI nor my unit test projects want to change that dependency (adding the runtime version tag to CSProj in the unit test project only updated the **displayed** runtime version in NuGet package manager - it still uses 2.1.0 and still fails).

Unit testing is simply broken for me for ASP.NET Core 2.1.x.

**Comment by dsplaisted** [comment]
For those who are still running into issues here, I do believe that adding `<TargetLatestRuntimePatch>true</TargetLatestRuntimePatch>` to the test project *should* fix the issue.  If this is not the case for you, it would probably be easiest for us to investigate what is going on if you file a new, separate issue, with as much repro detail as you can provide (including which version of the SDK you're using, what your project reference structure is like, and what errors you are getting).

Thanks!

**Comment by peterhuene** [comment]
I'd also like to understand the original repro a little bit more regarding why there's a need to specify a runtime identifier when building the unit test project (especially using `any` per the repro steps).

Is there a particular reason to specify a runtime identifier when building or running the unit tests?

**Comment by hof** [comment]
FWIW, I saw error NU1605 when building with `dotnet build -c Release -r win-x64` but *not* when building with `dotnet build -c Release`. 

After adding `<TargetLatestRuntimePatch>true</TargetLatestRuntimePatch>` building also worked with the `-r win-x64` flag. 

My project and test project both contain `<RuntimeIdentifiers>win-x64;linux-x64</RuntimeIdentifiers>`

**Comment by beefo** [comment]
Worked for me when I added `<TargetLatestRuntimePatch>true</TargetLatestRuntimePatch>` to both parent and unit test projects. Adding to test project alone did not work.

**Comment by tyconway** [comment]
I just ran into the same error as above;
Attempted to place <Target...>true<Target..> into the project file but got an error wherever I put it. (The suggestions to do so above by the way do not mention _where_ to put it; would anyone be willing to share?) 

I did resolve the error however by changing all the version numbers in the project file to the same number. That may be unwise--but it's worked so far.

**Comment by dsplaisted** [comment]
@tyconway It should go inside a `PropertyGroup`, similar to `TargetFramework`
